lumeer-remote/src/main/java/io/lumeer/remote/rest/AbstractService.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *  
+ * Copyright (C) since 2016 the original author or authors.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.lumeer.remote.rest;
+
+import io.lumeer.api.model.Resource;
+
+import java.net.URI;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.UriBuilder;
+
+public abstract class AbstractService {
+
+   @Inject
+   private HttpServletRequest request;
+
+   protected URI getResourceURI(Resource resource) {
+      return UriBuilder.fromUri(request.getRequestURL() + "/" + resource.getCode()).build();
+   }
+
+   protected URI getParentURI() {
+      String fullPath = request.getRequestURL().toString();
+      String parentPath = fullPath.substring(0, fullPath.lastIndexOf("/"));
```

Comment:
I am not sure that String manipulation is the smartest way of getting URI parts. I would prefer to rely on URI methods. There can be escaped slashes or some special occurrences to which this method does not apply. Like www.site.com/abc?p=a/b/c